docs/spec/v1beta1/imagerepositories.md

```diff
@@ -77,9 +77,17 @@ For a publicly accessible image repository, you will not need to provide a `secr
 
 When running in [<abbr title="Elastic Kubernetes Service">EKS</abbr>][EKS] and using [<abbr
 title="Elastic Container Registry">ECR</abbr>][ECR] to store images, you should be able to rely on
-the controller retrieving credentials automatically. The controller must be run with the flag
-`--aws-autologin-for-ecr` set for this to work. The advice under "Other platforms" below will also
-work for ECR.
+the controller retrieving credentials automatically.
+
+The `image-reflector-controller` must be run with the flag `--aws-autologin-for-ecr` set for this to work.
+
+This flag can be added by including a patch in the `kustomization.yaml` overlay file in your `flux-system`,
+as described in [Using Native AWS ECR Auto-Login][]. If there is no need for a security boundary on your
+cluster around container registries and you are not using Flux with so-called "soft multi-tenancy", then
+you will likely prefer to use the Auto-Login feature for the convenience and improved ease of use.
+
+Alternatively, the advice to use a cron job to refresh a secret token under [Other platforms][other platforms]
+below will also work with ECR for any environment that requires security boundaries and soft multi-tenancy.
```

Comment:
This might still be too much bootstrap-centric. If we prefer, I can shorten this to only say what was on line 82.

I'd still like to drop the link to `[Using Native AWS ECR Auto-Login][]` somewhere but otherwise think maybe could do without this entire paragraph line 84-87, it's important information but does it belong in the API doc?

I would merge it anyway if anyone says I'm being too picky, but still welcome feedback about this and anything else.

Reply:
Then again, for balance maybe this whole section goes away. We won't have a similar section for GCP or AKS here, I think (?)

Reply:
Since most of this information is already present under the ["Using Native AWS ECR Auto-Login" section](https://fluxcd.io/docs/guides/image-update/#using-native-aws-ecr-auto-login) in the "Automate image updates to Git" guide wouldn't it make sense to just extend that section and maybe just refer to it from here?

Reply:
I have adopted this suggestion in the latest version 👍